dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -132,6 +133,9 @@ private boolean scheduleNextTask() {
                 delegate.maybeExecute(components.endpoint(), components.request());
 
         if (response.isPresent()) {
+            // Complete the queued span
+            components.tracer().withTrace(() -> null);
```

Comment:
this isn't great

Reply:
there's also currently no way to properly create a span with the request path included in the span name. it's known here, but we have no way to pass that to the span created when the call is actually executed.